components/ui/shadcn-io/code-block/index.tsx

```
@@ -0,0 +1,620 @@
+    };
+
+    loadHighlightedCode();
+  }, [children, language, themes, syntaxHighlighting]);
+
+  if (!syntaxHighlighting || isLoading) {
+    return <CodeBlockFallback {...props}>{children}</CodeBlockFallback>;
+  }
+
+  return (
+    <div
+      dangerouslySetInnerHTML={{ __html: highlightedCode }}
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)